README.md

```diff
@@ -0,0 +1,211 @@
+```
+
+The output file will contain SSL certificate signature:
+```json
+{
+  "name" : "my.domain.com",
+  "fingerprint" : "jymEKdgGPv1zSp61CYya5c2fR9fTLe8tKnWF6857iLA=",
+  "expires" : 1543322263000,
+  "signature" : "MEUCICOs9bb6TIEmRNHCekxn9URADLYuuZnk4aftpVDzdwmWAiEAlU2r9VDEnAWryxvbAsSJfIlCQjKfumdFbZeUKda166w="
+}
+``` 
+
+Alternatively, you can prepare certificate signature from information about the certificate. This approach is useful when the certificate is not deployed yet.
+
+You need following information:
+* Domain common name, e.g. `my.domain.com`
+* Certificate fingerprint in HEX format, for example: `8f298429d8063efd734a9eb5098c9ae5cd9f47d7d32def2d2a7585ebce7b88b0`
+* SSL certificate expiration time as Unix timestamp, e.g. `1543322263000`
+
+This command will generate the SSL certificate signature:
+
+```sh
+java -jar ssl-pinning-tool.jar sign -k keypair.pem -f 8f298429d8063efd734a9eb5098c9ae5cd9f47d7d32def2d2a7585ebce7b88b0 -t 1543322263000 -n my.domain.com -o output.json -p [password]
+```
+
+The output file will contain JSON with SSL certificate signature:
+```json
+{
+  "name" : "my.domain.com",
+  "fingerprint" : "jymEKdgGPv1zSp61CYya5c2fR9fTLe8tKnWF6857iLA=",
+  "expires" : 1543322263000,
+  "signature" : "MEUCICOs9bb6TIEmRNHCekxn9URADLYuuZnk4aftpVDzdwmWAiEAlU2r9VDEnAWryxvbAsSJfIlCQjKfumdFbZeUKda166w="
+}
+``` 
+
+### Export Public Key
+
+Mobile app developers will need the public key from generated key pair in order to be able to verify signatures.
+
+You can convert EC private key to public key and print it:
+
+```sh
+java -jar ssl-pinning-tool.jar export -k keypair.pem -p [password]
+```
+
+### Troubleshooting
+
+Error: 
+```
+SEVERE: Failed to load private key, error: unable to read encrypted data: javax.crypto.BadPaddingException: pad block corrupted
+```
+
+This error is shown when the private key password is invalid.
+
+Alternatively, you can perform all steps above using `openssl` command.
+
+## Prepare Data Using OpenSSL
+
+In case you have your SSL certificate in a `pem` format, you are able to generate all necessary data using OpenSSL.
+
+### Generate a Key Pair
+
+This command will generate a new ECDSA key pair and store it in the `keypair.pem` file. You need to store this file securely.
+
+```sh
+openssl ecparam -name prime256v1 -genkey | openssl pkcs8 -topk8 -v2 aes-128-cbc > keypair.pem
+```
+
+**Store the key pair and private key password safely! You will need it next time you replace SSL certificate to generate new signatures.**
+
+### Prepare a Certificate Fingerprint
+
+This sequence of commands converts a server certificate stored in the `cert.pem` file to fingerprint encoded as Base64 and stored in `fingerprint.txt` file.
+
+```sh
+# Convert PEM to DER format
+openssl x509 -in cert.pem -inform PEM -outform DER -out cert.der
+
+# Compute SHA256 digest
+openssl dgst -sha256 < cert.der > fingerprint_raw.txt
+
+# Encode the digest as Base64
+openssl enc -base64 -A < fingerprint_raw.txt > fingerprint.txt
```

Comment:
These commands are slightly incorrect. We agreed with @hvge that we will convert the hex SHA-256 fingerprint into bytes and encode these bytes in Base64. Currently the hex value is encoded in Base64, which is different from the Java tool.